### DISCUSSION TITLE
Scaling XP based on the difference of player/enemy level for 1.14?

### DISCUSSION THREAD

**Discussion by WithinAmnesia** [flareteam/flare-game#890] [Q&A] [answered]
"Scaling XP based on the difference of player/enemy level is a reasonable idea." From https://github.com/flareteam/flare-engine/issues/1806#issuecomment-1012643094

@dorkster Short-form: Can this feature make it into 1.14 without too much hassle? If it can then maps with XP reward level scaling enemies become much more simplified to populate a map with from a game dev stand point. This feature can eliminate complex and convoluted map XP reward scaling level range enemy spawn testing and thus can increase game dev productivity by an order of magnitude.

Long-form: Currently I use 3 variants of the same monster that are manually XP reward level scaled and very carefully tested to ensure that the three different levels ranged controlled like an event spawner do not over lap (e.g 10% XP variant is spawned at player level 30 or above, 50% variant is spawned at player level 20+ and 100% XP is spawned at 10 or below). This is a complex work around and it is limited in successfully delivering upon player expectations. For as result of trying to manage complexity verses depth from a game dev content creation productively stand point; there tends to be an undesired outcome of a compromised and unintuitive / unnatural feeling 'XP cliffs' where a monster goes from 100% XP to 50% in the span of one level up. Also this makes 'low level farming' a gameplay experience eroding game design reality. As with 3 enemy XP reward level scaling brackets this forces the smarter players into elements of contrived yet efficient gameplay loops in order to become the most efficient while 'power leveling' with their limited time. Where as if there is a built in XP scaling feature for the enemies, than this effectively is like having every level difference become an XP scaling 'step'. As it then becomes possible from a realistic game dev work load and more over game design perspective to then gradually scale XP rewards up or down; depending on how close or far away a player's character level is compared to the possibly slain enemy level. As in one could go from 3 convoluted and error prone 3 effective XP reward brackets for enemies with clunky game dev work to implement. To anywhere from 20 or 30 or more intuitive and natural feeling scaling XP reward level scaling brackets / 'steps' with far less hassle from a game dev's perspective to implement. Albeit I am just not with a great degree of clarity and or coding understanding as to how difficult and or how complex implementing XP level scaling rewards are for the Fl-are engine/ possibly Flare 1.14?

**Comment by dorkster** [comment, marked as answer]
I don't think the implementation would be too complicated, so I'll go ahead and say "yes" at the moment.
